### PR TITLE
Clean up sbt config

### PR DESCRIPTION
**Pull request by timcharper**
Summary:
- Restore MAINTAINER label for our Docker image
- Remove left over NativePackager code
- Remove references to jdeb and associated workaround
- Remove sbt-in-sbt loading of Coursier (this causes an issue, see https://github.com/sbt/sbt-native-packager/issues/1053#issuecomment-424118860)
- Sync sbt-in-sbt version

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -165,9 +165,11 @@ lazy val packagingSettings = Seq(
     ("./version docker" !!).trim
   },
   (defaultLinuxInstallLocation in Docker) := "/marathon",
+  maintainer := "Mesosphere Package Builder <support@mesosphere.io>",
   dockerCommands := {
-    // kind of a work-around; we want our mesos install and jdk install to come earlier so that Docker can cache them
-    val (prefixCommands, restCommands) = dockerCommands.value.splitAt(2)
+    // kind of a work-around; we need our chown /marathon command to come after the WORKDIR command, and installation
+    // commands to preceed adding the Marthon artifact so that Docker can cache them
+    val (prefixCommands, restCommands) = dockerCommands.value.splitAt(3)
```
Comment (alenkacz) [inline build.sbt]: I don't understand why splitAt(2) changed to splitAt(3) :/

Comment (timcharper) [reply build.sbt]: >  we need our chown /marathon command to come after the WORKDIR command

:)

Comment (timcharper) [reply build.sbt]: I thought about switching it to looking for the WORKDIR command; that would be clearer, but it was more changes.

Comment (timcharper) [reply build.sbt]: Is this better for you?

```scala
    val (prefixCommands, restCommands) = dockerCommands.value.splitAt(dockerCommands.value.indexWhere(_.makeCommand.startsWith("WORKDIR ")) + 1)
```

Comment (alenkacz) [reply build.sbt]: That's absolutely perfect 😍

**Pull request merged**